crates/sui-bridge-indexer/src/eth_bridge_indexer.rs

```diff
@@ -294,6 +267,127 @@ impl Datasource<RawEthData> for EthSyncDatasource {
     }
 }
 
+async fn retrieve_and_process_live_finalized_logs(
+    client: Arc<EthClient<MeteredEthHttpProvier>>,
+    provider: Arc<Provider<Http>>,
+    addresses: Vec<EthAddress>,
+    starting_checkpoint: u64,
+    data_sender: DataSender<RawEthData>,
+    bridge_metrics: Arc<BridgeMetrics>,
+) {
+    let eth_contracts_to_watch = HashMap::from_iter(
+        addresses
+            .iter()
+            .map(|address| (*address, starting_checkpoint)),
+    );
+
+    let (_, mut eth_events_rx, _) = EthSyncer::new(client.clone(), eth_contracts_to_watch)
+        .run(bridge_metrics.clone())
+        .await
+        .expect("Failed to start eth syncer");
+
+    // forward received events to the data sender
+    while let Some((_, block, logs)) = eth_events_rx.recv().await {
+        let raw_logs: Vec<RawEthLog> = logs
+            .into_iter()
+            .map(|log| RawEthLog {
+                block_number: block,  // Handle block number conversion
+                tx_hash: log.tx_hash, // Assuming `event` contains `tx_hash`
+                log: log.log,
+            })
+            .collect();
+
+        process_logs(raw_logs, provider.clone(), data_sender.clone(), block)
+            .await
+            .expect("Failed to process logs");
```

Comment:
do we really want to panic here? (we didn't panic in `retrieve_and_process_log_range`)

Reply:
I did this because we aren't returning a result from the live task, like the sync task does.

Reply:
not sure i follow. `retrieve_and_process_log_range` (is this what you meant by "sync task"?) does not return anything. One of them is `process_logs().await.expect()` and the other one is `process_logs().await?`

Reply:
Yes. My idea was to not have `retrieve_and_process_live_finalized_logs` return a result as the task should never die.